CONTRIBUTING.md

```diff
@@ -11,6 +11,53 @@ composer install && ./vendor/bin/phpunit
 -   [Make a pull request](https://help.github.com/articles/using-pull-requests)
 -   Thanks!
 
+Testing
+=======
+
+Unit tests
+----------
+
+To run the unit tests, install the dependencies with Composer:
+
+```sh
+$ composer install
+```
+
+Then run the Composer "test" script:
+
+```sh
+$ composer test
+```
+
+End-to-end tests
+----------------
+
+These tests are implemented with our notifier testing tool [Maze Runner](https://github.com/bugsnag/maze-runner). This requires a recent version of Ruby to run
+
+End to end tests are written in cucumber-style `.feature` files, and need Ruby-backed "steps" in order to know what to run. The tests are located in the top level [`features`](/features/) directory
+
+Install Maze Runner using Bundler:
+
+```sh
+$ bundle install
+```
+
+Configure the tests with the following environment variables:
+
+- `PHP_VERSION` — the PHP version to run the tests with, e.g. "8.0". This must match one of the [published PHP Docker image tags](https://hub.docker.com/_/php)
+- `SYMFONY_VERSION` — the version of Symfony to run the tests against, e.g. "5". There must be a matching fixture for the specified version in the [`features/fixtures`](/features/fixtures/) directory
+
+Run Maze Runner with Bundler:
+
+```sh
+$ PHP_VERSION=8.0 SYMFONY_VERSION=5 bundle exec maze-runner
```

Comment:
I guess I'm missing a dependency or some setup, but when I try this all the scenarios seem to fail:
```
D, [2021-07-08 22:32:46#7980] DEBUG -- : docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
D, [2021-07-08 22:32:46#7980] DEBUG -- : [8002] Failed to execute script docker-compose
```

Reply:
There's no extra setup when it runs on CI — https://github.com/bugsnag/bugsnag-symfony/blob/test/add-maze-runner/.github/workflows/maze-runner.yml

Can you get any more output than that? e.g. which file is missing

Reply:
Oh could this be because I used the latest docker-compose version?

https://github.com/bugsnag/bugsnag-symfony/blob/55548dc48ea852870ff8896d8a1e12525d474111/features/fixtures/docker-compose.yml#L1

There's no real reason to, so it could be dropped but that would explain it if your docker version is too old

Reply:
I just didn't have Docker running, after closing it by mistake when it was nagging to update.